examples: fix link order in librados example Makefile

MahatiC opened the pull request:
The library link order in librados example Makefile is incorrect
and as a result 'make' throws an error. This change fixes the order.

Signed-off-by: Mahati Chamarthy <mahati.chamarthy@intel.com>

tchaikov commented:
the Makefile works for me.
```
examples/librados|master|$ make
g++ -std=c++11 -Wall -Wextra -Werror -g -I../../src/include -L../../build/lib/ -Wl,-rpath,../../build/lib -lboost_system -lrados -lradosstriper -o hello_world_cpp hello_world.cc
g++ -std=c++11 -Wall -Wextra -Werror -g -I../../src/include -L../../build/lib/ -Wl,-rpath,../../build/lib -lboost_system -lrados -lradosstriper -o hello_radosstriper_cpp hello_radosstriper.cc
cc -Wall -Wextra -Werror -g -I../../src/include -L../../build/lib/ -Wl,-rpath,../../build/lib -lrados -o hello_world_c hello_world_c.c
```

@MahatiC could you share with us what error was thrown by `make`? and what was the command line you used to launch `make`?

tchaikov commented:
also could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes, in this case, it would be "examples:".

MahatiC commented:
@tchaikov thanks for your review. I updated the commit title. Here is the error I receive when I invoke 'make' https://pastebin.com/RgWrsqfR

Did you have any paths set in your environment? My understanding of the error was that, per this https://gcc.gnu.org/onlinedocs/gcc-5.2.0/gcc/Link-Options.html , the libraries referred in "hello_world.cc" are _before_ the file "hello_world.cc", hence the link error. So when I put them towards the end in g++ command order, it didn't throw any error.

Note: I have now changed the code a bit, to make it more readable. Thanks.

tchaikov commented:
> Did you have any paths set in your environment? 

other than `$PATH`, no path is set my building env.

if `CXX_LIBS?=-lboost_system -lrados -lradosstriper` is commented out, i have the very same link error as yours.

MahatiC commented:
> i reproduces your issue with gcc 4.8.4 shipped with ubuntu trusty. and you patch addresses the issue. but it builds fine with gcc 7.2 without this change.

Looks like that's the issue. Thanks.

> i am not sure how you connect the doc to the solution and the error ? could you point me to the specific line that supports your theory?

I was referring to this line in the doc
> It makes a difference where in the command you write this option; the linker searches and processes libraries and object files in the order they are specified. Thus, ‘foo.o -lz bar.o’ searches library ‘z’ after file foo.o but before bar.o. If bar.o refers to functions in ‘z’, those functions may not be loaded.

On a diff note: While working on librbd, I created an example with librbd API, similar to the librados example in the tree. Would it be of any value to propose that example upstream under /examples? Thank you.
